### PR TITLE
#2485. Add `is! T` constant evaluation tests

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: Language/Expressions/Constants/is_not_type_A01_t02.dart [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+///   prefix, optionally followed by type arguments of the form <T1, ..., Tn>
+///   where T1, ..., Tn are constant type expressions.
+/// • A type of the form FutureOr<T> where T is a constant type expression.
+/// • A function type R Function<typeParameters>(argumentTypes) (where
+///   R and <typeParameters> may be omitted) and where R, typeParameters
+///   and argumentTypes (if present) contain only constant type expressions.
+/// • The type void.
+/// • The type dynamic
+///
+/// @description Checks that an expression of the form `e is T` is a constant
```
Comment (eernstg) [inline Language/Expressions/Constants/is_not_type_A01_t02.dart:24]: Perhaps:
```suggestion
/// @description Checks that an expression of the form `e is! T` is a constant
```

File: Language/Expressions/Constants/is_not_type_A01_t03.dart [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+///   prefix, optionally followed by type arguments of the form <T1, ..., Tn>
+///   where T1, ..., Tn are constant type expressions.
+/// • A type of the form FutureOr<T> where T is a constant type expression.
+/// • A function type R Function<typeParameters>(argumentTypes) (where
+///   R and <typeParameters> may be omitted) and where R, typeParameters
+///   and argumentTypes (if present) contain only constant type expressions.
+/// • The type void.
+/// • The type dynamic
+///
+/// @description Checks that an expression of the form `e is T` is a constant
```
Comment (eernstg) [inline Language/Expressions/Constants/is_not_type_A01_t03.dart:24]: Similar change:
```suggestion
/// @description Checks that an expression of the form `e is! T` is a constant
```

File: Language/Expressions/Constants/is_not_type_A02_t01.dart [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A constant expression is an expression whose value can never
+/// change, and that can be evaluated entirely at compile time.
+/// A constant expression is one of the following:
+/// ...
+/// • An expression of the form e is! T is equivalent to !(e is T) in every
+///   way, including whether it’s potentially constant or constant.
+///
+/// @description Checks that an expression of the form `e is T` is not a
```
Comment (eernstg) [inline Language/Expressions/Constants/is_not_type_A02_t01.dart:12]: Ditto
```suggestion
/// @description Checks that an expression of the form `e is! T` is not a
```

File: Language/Expressions/Constants/is_not_type_A02_t02.dart [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+// [cfe] unspecified
+  const Void = void is! Type;
+//             ^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  const cIntExtension = IntExtension is! Type;
+//                      ^^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  const F = () {} is! Function;
```
Comment (eernstg) [inline Language/Expressions/Constants/is_not_type_A02_t02.dart:31]: This one _is_ an expression, just not a constant one. Right?

**Review by eernstg — COMMENTED** [review]
Looks good! Couple of typos.

**Comment by sgrekhov** [comment]
Updated. Please take another look

**Review by eernstg — APPROVED** [review]
LGTM!

**Pull request merged**